### ISSUE TITLE
PHP7 Support

### ISSUE THREAD

**Issue by cubiclelord** [mayflower/puppet-php#146]
Hi, 

I'm trying to configure PHP7 with this module (we're already using it with PHP5), and I'm seeing a few places that hardcode in php5. Is there any plan to adjust these settings so it will work correctly with PHP7?  

Thanks!

**Comment by fpletz** [comment]
Yes, we really should support PHP7 in the near future. I'm not sure though if we will be able to do this in the next few weeks. Pull requests are always welcome if you would like to take a shot at it.

**Comment by cgalvarez** [comment]
Hi, cubiclelord. I didn't see this issue before. I had the same problem, and made a fork and the subsequent pull request #147. Yesterday I managed it to pass all existing tests. Of course, it installs PHP 7 (I'm using it with WordPress in my virtual machine).

You can try it while waiting for it to be merged into the master branch. In the pull request description you have examples on how I set my hiera file up to get it working.

**Comment by mmoll** [comment]
There's still work to do in the php-version-select for this to work without setting multiple of parameters just for the version, but it can be done now that way.

**Comment by pkordylewski** [comment]
Is there an ETA for PHP7 support?

Thanks

**Comment by mmoll** [comment]
I guess this would have to come from "the community" now.

**Comment by fpletz** [comment]
We do not have full PHP7 support, so reopen. The should probably be easily switchable by a parameter to the main class which would set the correct parameters for locations etc.

**Comment by FutureSharks** [comment]
On Ubuntu 16.0 I was able to get PHP 7 working with these settings:

php::manage_repos: false
php::params::cfg_root: /etc/php/7.0
php::fpm::config::pid_file: /var/run/php/php7.0-fpm.pid
php::fpm::config::error_log: /var/log/php7.0-fpm.log
php::fpm_service_name: php7.0-fpm
php::ext_tool_enable: /usr/sbin/phpenmod
php::ext_tool_query: /usr/sbin/phpquery
php::package_prefix: php-

Ideally these settings would come from the params class. I could make a PR when I have some time.

**Comment by max-rocket-internet** [comment]
I've submitted a PR for PHP7 support for Ubuntu/Debian:
https://github.com/mayflower/puppet-php/pull/180
Let me know what you think.

**Comment by FutureSharks** [comment]
I've submitted a PR for PHP7 support for Ubuntu/Debian:
https://github.com/mayflower/puppet-php/pull/180
Let me know what you think.

**Comment by FutureSharks** [comment]
Any update? Would be great to get my PR merged. Other people who are using RPM based distros could contribute to the params class too, to cover those OSs.

**Comment by mmoll** [comment]
@fpletz could you please communicate our results and open the needed issues to outline the way(s) to go?

**Comment by FutureSharks** [comment]
@fpletz any update?

**Comment by fpletz** [comment]
I've finally gotten around to write the ticket for finding new maintainers: #183 

I will review your PR shortly. Thanks!

**Comment by craigwatson** [comment]
@FutureSharks I've gotten this module working with PHP 7 (Ubuntu 16.04) thanks to your parameters, however I had to modify `php::ext_tool_query: /usr/sbin/phpquery` to `php::ext_tool_query: '/usr/sbin/phpquery -v 7.0'`

**Comment by max-rocket-internet** [comment]
@craigwatson 
After posting those params I actually had to just disable the query tool because it looks like its exit code is not correct in Ubuntu 16.04 and causes the modules to not install.

I set this in hiera:
`php::ext_tool_enabled: false
`

On Ubuntu 14.04:
```
# php5query -s cli -m curl; echo $?; php5query -s cli -m blahblah; echo $?
curl (Enabled for cli by maintainer script)
0
No module matches blahblah
1
```

On Ubuntu 16.04:
```
root@testing-vendorbackend-0bf9586c740a7e35a:in:~# phpquery -v 7.0 -s cli -m curl; echo $?; phpquery -v 7.0 -s cli -m blahblah; echo $?
curl (Enabled for cli by maintainer script)
0
No module matches blahblah
0
```

**Comment by FutureSharks** [comment]
@craigwatson 
After posting those params I actually had to just disable the query tool because it looks like its exit code is not correct in Ubuntu 16.04 and causes the modules to not install.

I set this in hiera:
`php::ext_tool_enabled: false
`

On Ubuntu 14.04:
```
# php5query -s cli -m curl; echo $?; php5query -s cli -m blahblah; echo $?
curl (Enabled for cli by maintainer script)
0
No module matches blahblah
1
```

On Ubuntu 16.04:
```
root@testing-vendorbackend-0bf9586c740a7e35a:in:~# phpquery -v 7.0 -s cli -m curl; echo $?; phpquery -v 7.0 -s cli -m blahblah; echo $?
curl (Enabled for cli by maintainer script)
0
No module matches blahblah
0
```

**Comment by craigwatson** [comment]
@FutureSharks good catch!

**Comment by JackBlower** [comment]
Will the PHP 7 support support installing 5.6 and 7 side-by-side? (Like Ondrej's PHP 7 repo supports)

**Comment by fadenb** [comment]
@JackBlower there are no plans to implement such functionality.

**Comment by mmoll** [comment]
Could all interested parties please test the current master with php_version set, as described in the README and report back? (pull requests in case of problems welcome, of course).

**Comment by craigwatson** [comment]
@mmoll the value [here](https://github.com/mayflower/puppet-php/blob/master/manifests/globals.pp#L26) is wrong if you're using the OS defaults rather than a PPA.

PR incoming :)

**Comment by kenyon** [comment]
Done in #180 and #220.
